Added sort support

CaselIT opened the pull request:
Add support for sorting the return values of a `SQLAlchemyConnectionField`

The `sort` argument of `SQLAlchemyConnectionField` supports sorting over a single over multiple field.

In the utils file I've added
-  the function `sort_enum_for_model` to create a graphene Enum with all be possible field and order combination for a particular sqlalchemy model as suggested by @Cito 
- the function `sort_argument_for_model` to automacally create the enum and return the attribute to use in the connection field

The test and documentation are still missing, because first I wanted to get some feedback about the implementation.

Regarding the value of the sort enumerator at first I tried to set the values of the enum to the ascenting or descenting operators of sqlalchemy simplify the logic of the get_query but that results in an error when graphql builds the ast graph, so I reverted to use a list column_name, operator that is less than optimal since every time it needs to query the model to create the operators.


Thanks to @Cito to the input in #40 
Closes #40

CaselIT commented:
> Regarding the value of the sort enumerator at first I tried to set the values of the enum to the ascenting or descenting operators of sqlalchemy simplify the logic of the get_query but that results in an error when graphql builds the ast graph, so I reverted to use a list column_name, operator that is less than optimal since every time it needs to query the model to create the operators.
I've changed the implementation to use a subclass of string instead of the list, so that the ast graph can be created successfully and there is no need to query the model every time.

Cito commented:
Nice, thank you.

Maybe, as I've already commented in #40, the `sort` argument should be added by default to the `SQLAlchemyConnectionField`, so that you don't need to specify it manually. That would be similar to how a Graphene `ConnectionField` comes with `first`, `last` arguments by default, which also don't need to added manually.

CaselIT commented:
Good idea. I'll add it.
Do you think it should have a default value when added by default?

Cito commented:
For an `SQLAlchemyConnectionField` it probably makes sense to use the `id` as default value for the sort argument so that you always get a reproducable order.

CaselIT commented:
`SQLAlchemyConnectionField` now adds the sort attribute by default, with a default sort over the primary key(s). To disable this field pass to the constructor `sort=None`

Kazanz commented:
This going to get merged?

CaselIT commented:
I home so.
@syrusakbary could you review this PR?
Thanks

CaselIT commented:
@syrusakbary see https://github.com/graphql-python/graphene-sqlalchemy/pull/104#issuecomment-361208698

YehonatanVonage commented:
Hi, 
Any updates about this PR?

CaselIT commented:
On my part I wanted a feedback from @syrusakbary about the implementation before writing the tests.

millicentachieng commented:
@syrusakbary We are waiting on you to review the PR

syrusakbary commented:
Sorry for the time taken on reviewing this PR, and thanks for the ping.
It looks good, once it have tests related to the new functionality I'll merge :)

CaselIT commented:
Thanks for the review @syrusakbary 
I'll work on adding the tests and ping you once they are done

CaselIT commented:
@syrusakbary I think it's ready for a review
I've added the tests
Updated to support graphene v2
Added a documentation example and updated some other documentation to be compatible with v2

Let me know what you think

syrusakbary commented:
It looks great! You did a great job 👏 

Merging

CaselIT commented:
Thank you for merging it!
